ledger/src/bank_forks.rs

```diff
@@ -177,35 +177,50 @@ impl BankForks {
             .last()
             .map(|bank| bank.transaction_count())
             .unwrap_or(0);
-
-        root_bank.squash();
-        let new_tx_count = root_bank.transaction_count();
-
-        // Generate a snapshot if snapshots are configured and it's been an appropriate number
-        // of banks since the last snapshot
+        // Generate each snapshot at a fixed interval
         if self.snapshot_config.is_some() && snapshot_package_sender.is_some() {
             let config = self.snapshot_config.as_ref().unwrap();
-            info!("setting snapshot root: {}", root);
-            if root - self.last_snapshot_slot >= config.snapshot_interval_slots as Slot {
-                let mut snapshot_time = Measure::start("total-snapshot-ms");
-                let r = self.generate_snapshot(
-                    root,
-                    &root_bank.src.roots(),
-                    snapshot_package_sender.as_ref().unwrap(),
-                    snapshot_utils::get_snapshot_archive_path(&config.snapshot_package_output_path),
-                );
-                if r.is_err() {
-                    warn!("Error generating snapshot for bank: {}, err: {:?}", root, r);
-                } else {
-                    self.last_snapshot_slot = root;
+            let mut banks = vec![root_bank];
+            let parents = root_bank.parents();
+            banks.extend(parents.iter());
+            for bank in banks {
+                let parent_slot = bank.parent_slot();
+                let root = bank.slot();
+                if Self::check_interval(root, parent_slot, config.snapshot_interval_slots) {
```

Comment:
If the snapshot slots would be synchronized across the cluster, there might be noticeable hiccups especially because all leaders at the time of snapshotting is locked away from any bank while bookkeeping for snapshot.

This is might not be problem now. But, if there are huge active accounts, it might be problem?

Reply:
Yep.  we will see.  I think it can be asynchronous and hopefully not impact the cluster to much.